aten/src/ATen/native/xnnpack/ChannelShuffle.cpp

```diff
@@ -0,0 +1,104 @@
+#ifdef USE_XNNPACK
+
+#include <ATen/native/xnnpack/Common.h>
+#include <ATen/native/utils/Factory.h>
+
+namespace at {
+namespace native {
+namespace xnnpack {
+
+bool use_channel_shuffle(
+    const Tensor& input,
+    const int64_t groups) {
+  using namespace internal;
```

Comment:
Why do you need this (`using namespace internal`)?

Reply:
That's for using `Layout::Activation4D` in the function. I think `MaxPooling.cpp` did the same thing.

Reply:
Oh I see. OK thats fine.